### PR TITLE
Finish PauliStringsJax

### PR DESCRIPTION
**Pull request by inailuig**
Stuff which didn't make it into https://github.com/netket/netket/pull/1510 before it was merged.
I had done two implementations of the operator (mask and index, and you could do sth hybrid), this 
- tests both
- switches the default to indexing, which seems faster for short strings (on gpu one needs to test)
In particular with indexing it almost matches the performance of isingjax (convert it with -> to_local_operator->to_pauli_strings->to_jax_operator)
- privately exposes choosing the implementation in the constructor of the operator

### CODE REVIEW

**Review by PhilipVinc — APPROVED** [review]

**Comment by PhilipVinc** [comment]
can you add a commend on what the two modes do and how they differ?

**Comment by PhilipVinc** [comment]
next to where it is stored

File: netket/operator/_pauli_strings/jax.py [hunk 1]
```diff
@@ -304,8 +315,16 @@ def __init__(
                 "nonzero cuttof is not yet implemented in PauliStringsJax"
             )
         # private variable for setting the mode
+        # currently there are two modes:
+        # 1. index: indexes into the vector to flip qubits and compute the sign
```
Comment (inailuig) [inline netket/operator/_pauli_strings/jax.py:319]:
```suggestion
        # index: indexes into the vector to flip qubits and compute the sign
```

File: netket/operator/_pauli_strings/jax.py [hunk 2]
```diff
@@ -304,8 +315,16 @@ def __init__(
...
+        #           faster if the strings act only on a few qubits
+        # 2. mask: uses masks to flip qubits and compute the sign
```
Comment (inailuig) [inline netket/operator/_pauli_strings/jax.py:321]:
```suggestion
        # mask: uses masks to flip qubits and compute the sign
```
